### ISSUE TITLE
1.11.0: glob pattern in globalDependencies

### ISSUE THREAD

**Issue by mistic100** [vercel/turbo#6729]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

https://github.com/mistic100/Photo-Sphere-Viewer/tree/dependabot/npm_and_yarn/turbo-1.11.0

### What package manager are you using / does the bug impact?

Yarn v1

### What operating system are you using?

Linux

### Which canary version will you have in your reproduction?

demo uses 1.11.0 but bug is present in 1.10.17-canary.10 too

### Describe the Bug

I have a bunch of build tools in the ["build" directory of my project and it's subfolders](https://github.com/mistic100/Photo-Sphere-Viewer/tree/dependabot/npm_and_yarn/turbo-1.11.0/build). It is mix of .js and .mjs files (for now, it could change).

My turbo.json contains :

```json
{
  "$schema": "https://turborepo.org/schema.json",
  "globalDependencies": [
    "build/**"
  ],
}
```

This worked well on 1.10.0 and before but since 1.11.0 this fails because it tries to hash one of the subdirectory

> ERROR  run failed: git error on /home/..../Photo-Sphere-Viewer/build/plugins: error reading file for hashing: Is a directory; class=Os (2)

I can change my config to this but it is not future proof.

```json
{
  "$schema": "https://turborepo.org/schema.json",
  "globalDependencies": [
    "build/**/*.js",
    "build/**/*.mjs",
  ],
}
```

### Expected Behavior

I think the glob resolution should filter out directories before passing the list to the hasher.

### To Reproduce

- Checkout https://github.com/mistic100/Photo-Sphere-Viewer/tree/dependabot/npm_and_yarn/turbo-1.11.0
- run yarn
- run yarn ci:build

**Comment by arlyon** [comment]
We have fixed the issue and are planning on releasing a new canary with these changes shortly.

**Comment by psychobolt** [comment]
Is there support for a dot in the input path? I don't believe the fix handles the case when there is a dot in the input path e.g.
```json
  "globalDependencies": [
    ".storybook/**"
  ],
```

When I created a empty root task input, it works correctly:
```json
  "pipeline": {
    "//#globals": {
      "inputs": [".storybook/**"]
    },
    "build": {
      "dependsOn": ["//#globals", "^build"]
    }
  }
```

**Comment by arlyon** [comment]
Hi @psychobolt could you do me a favour and open a new issue with these cases and a reproduction repo? I have received a report similar to this already but am still waiting on a reproduction as I couldn't build one from scratch.
